File: AndroidStudioProjects/yogainstructorapp/app/src/main/java/com/example/yogainstructorapp/data/DatabaseProvider.java
package com.example.yogainstructorapp.data;

import android.content.Context;
import androidx.room.Room;
import androidx.room.RoomDatabase;

@Database(entities = {User.class}, version = 1)
public abstract class AppDatabase extends RoomDatabase {
    public abstract UserDao userDao();
}

public class DatabaseProvider {
    private static AppDatabase database;

    public static AppDatabase getDatabase(final Context context) {
        if (database == null) {
            synchronized (AppDatabase.class) {
                if (database == null) {
                    database = Room.databaseBuilder(context.getApplicationContext(),
                            AppDatabase.class, "yoga_database").build();
                }
            }
        }
        return database;
    }
}

File: AndroidStudioProjects/yogatry1/data/data/DatabaseProvider.kt
package com.example.yogainstructorapp.data

import androidx.room.Room
import androidx.room.RoomDatabase
import android.content.Context

object DatabaseProvider {
    private var INSTANCE: AppDatabase? = null

    fun getDatabase(context: Context): AppDatabase {
        return INSTANCE ?: synchronized(this) {
            val instance = Room.databaseBuilder(
                context.applicationContext,
                YourDatabase::class.java,
                "your_database_name"
            ).build()
            INSTANCE = instance
            instance
        }
    }
}
